libc/docs/dev/code_style.rst

```diff
@@ -186,3 +186,32 @@ We expect contributions to be free of warnings from the `minimum supported
 compiler versions`__ (and newer).
 
 .. __: https://libc.llvm.org/compiler_support.html#minimum-supported-versions
+
+Policy on Assembler sources
+===========================
+
+Coding in high level languages such as C++ provides benefits relative to low
+level languages like Assembler, such as:
+
+* Improved safety
+* Instrumentation
+
+  * Code coverage
+  * Profile collection
+* Sanitization
+* Debug info
+
+While its not impossible to have Assembler code that correctly provides all of
+the above, we do not wish to maintain such Assembler sources in llvm-libc.
+
+That said, there a few functions provided by llvm-libc that are more difficult
+to implement or maintain in C++ than Assembler. We do use inline or out-of-line
```

Comment:
```suggestion
That said, there a few functions provided by llvm-libc that are impossible
to reliably implement in C++. We do use inline or out-of-line
```

Reply:
Maybe add, "...to reliably implement in C++ for all compilers supported for building the libc."

Reply:
yeah, that's a point worth making explicit --- functions like this were hard enough for bionic to deal with, and we only had to support gcc _or_ clang at any given time. just the different versions of the _same_ compiler were painful enough, without having to worry about two completely different compilers. (and we did junk several such functions as we transitioned from gcc to clang!)

Reply:
I've reworded this in c06aa9d3d5ee to:

> That said, there a few functions provided by llvm-libc that are impossible
to reliably implement in C++ for all compilers supported for building
llvm-libc.